tests/farm/LiquidityMining.js

```diff
@@ -677,64 +833,553 @@ contract("LiquidityMining", (accounts) => {
             });
         });
 
-        it("should be able to deposit using wrapper", async () => {
-            let tx = await liquidityMining.deposit(token1.address, amount, account2, {
-                from: account1,
-            });
+        it("should be able to deposit using wrapper", async () => {
+            let tx = await liquidityMining.deposit(token1.address, amount, account2, {
+                from: account1,
+            });
+
+            let poolInfo = await liquidityMining.getPoolInfo(token1.address);
+            let blockNumber = new BN(tx.receipt.blockNumber);
+            checkPoolInfo(poolInfo, token1.address, allocationPoint, blockNumber, new BN(0));
+
+            await checkUserPoolTokens(account2, token1, amount, amount, new BN(0));
+
+            expectEvent(tx, "Deposit", {
+                user: account2,
+                poolToken: token1.address,
+                amount: amount,
+            });
+        });
+
+        it("fails if token pool token not found", async () => {
+            await expectRevert(
+                liquidityMining.deposit(account1, amount, ZERO_ADDRESS, { from: account1 }),
+                "Pool token not found"
+            );
+        });
+    });
+
+    describe("claimReward", () => {
+        let allocationPoint = new BN(1);
+        let amount = new BN(1000);
+
+        beforeEach(async () => {
+            await deploymentAndInit();
+            await liquidityMining.add(token1.address, allocationPoint, false);
+            await mineBlocks(1);
+
+            await token1.mint(account1, amount);
+            await token1.approve(liquidityMining.address, amount, { from: account1 });
+        });
+
+        it("shouldn't be able to claim reward (will not be claimed without SOV tokens)", async () => {
+            await liquidityMining.deposit(token1.address, amount, ZERO_ADDRESS, {
+                from: account1,
+            });
+
+            await expectRevert(
+                liquidityMining.claimReward(token1.address, ZERO_ADDRESS, { from: account1 }),
+                "Claiming reward failed"
+            );
+        });
+
+        // if claiming rewards for single pool and 0% <unlockedImmediatelyPercent<100% and poolTokensUnlockedImmediatelyPercent is 0, unlockedImmediatelyPercent of the reward amount gets transferred to the user directly (through the lockedSOV contract) and 100% - unlockedImmediatelyPercent are vested
+        it("should be able to claim reward (will be claimed with SOV tokens) with 10% of unlockedImmediatelyPercent", async () => {
+            /**
+             * unlockedImmediatelyPercent was set to 10%
```

Comment:
recommendation: it is more clean to have unit tests isolated - running a setup() fixture on each test